modules/command-engine/core/src/main/scala/surge/internal/domain/SurgeMessagePipeline.scala

```diff
@@ -70,7 +70,7 @@ private[surge] abstract class SurgeMessagePipeline[S, M, E](
   protected val cqrsRegionCreator: PersistentActorRegionCreator[M] =
     new PersistentActorRegionCreator[M](actorSystem, businessLogic, kafkaStreamsImpl, partitionTracker, businessLogic.metrics, signalBus, config = config)
 
-  protected val actorRouter: SurgePartitionRouter =
+  protected lazy val actorRouter: SurgePartitionRouter =
```

Comment:
Any reason in particular to make this lazy?

Reply:
In the SurgeMessagePipelineSpec, when we override the actorRouter it is evaluated twice. Lazy val resolves that issue.